test.js

```diff
@@ -148,3 +149,21 @@ test('#27, does not remove spaces in line with ansi escapes when no trimming', t
 	t.is(wrapAnsi(chalk.bgGreen(`  ${chalk.black('OK')} `), 100, {trim: false}), chalk.bgGreen(`  ${chalk.black('OK')} `));
 	t.is(wrapAnsi(chalk.bgGreen(' hello '), 10, {hard: true, trim: false}), chalk.bgGreen(' hello '));
 });
+
+test('#35, wraps hyperlinks, preserving clickability in supporting terminals', t => {
+	const link = 'https://testlogin:testpassword@areallylongurlthatneedstogetwrapped.com/with_some/path?andparams=bogus&more=evenmorebogus#evensomehashparams=notdoinganything';
```

Comment:
It would be nice if the text in test cases was much shorter. At the moment it's hard to even figure out what is happening.

Reply:
UPD: In the end, I failed to understand what is happening in the end so didn't review the test cases.

Reply:
Hopefully I clarified things a bit. Let me know if it's still incomprehensible though.

Reply:
Probably could use splitting the lines in expected too. But after some editing i can see the logic now.

Anyways, I tried this in my terminal(`termite`) and on a couple others(from supported list at https://github.com/sindresorhus/terminal-link, I tried `guake` and `terminology`), input works, output not clickable on lines beyond first, not displayed on lines beyond first, a click on first line results in opening incorrect link.

Reply:
Also, after printing this, all terminals entered state where nothing that is printed is displayed at all(until reset).

Reply:
After having tested in a number of terminal emulators, it seems that many don't even support ansi links. I did a simple test using terminal-link in DomTerm, Terminology, and iTerm2 (which are all stated to support it), as well as Hyper and Termite (which are noted to have feature requests sent) and Terminus. Only iTerm2 exhibited the correct behaviour. I'm not why DomTerm and Terminology didn't work... I wasn't able to build guake to test.

In any case, I think I know the reason printing the output breaks the terminal display. I will work on this.